src/openforms/contrib/kadaster/api/views.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import re
+
+from django.conf import settings
+from django.http import HttpResponseBadRequest
+from django.utils.translation import gettext_lazy as _
+
+import requests
+from djangorestframework_camel_case.render import CamelCaseJSONRenderer
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
+from openforms.submissions.api.permissions import AnyActiveSubmissionPermission
+
+from .serializers import MapSearchSerializer
+
+logger = logging.getLogger(__name__)
+
+
+@extend_schema(
+    summary=_("List BAG address suggestions."),
+    parameters=[
+        OpenApiParameter(
+            "q",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("The search query we send to the pdok locatieserver api."),
+            required=True,
+        )
+    ],
+    responses={
+        200: MapSearchSerializer,
+        status.HTTP_400_BAD_REQUEST: ValidationErrorSerializer,
+        status.HTTP_401_UNAUTHORIZED: ExceptionSerializer,
+    },
+)
+class MapSearchView(GenericAPIView):
```

Comment:
I think you can considerably remove boilerplate by using https://open-forms.readthedocs.io/en/stable/developers/backend/core/utils.html#openforms.api.views.mixins.ListMixin so you only have to implement `get_objects` and don't have to deal with the serializer/response boilerplate

Reply:
What do you exactly mean by boiler plating? Because the only thing I can think of when I see that is the spectacular code, which I think is good to have for documentation purposes.